### ISSUE TITLE
ExternalDocumentRef with non sha1 checksums do not validate correctly

### ISSUE THREAD

**Issue by puerco** [spdx/tools-java#21]
When validating a spdx Tag/Value doc with an `ExternalDocumentRef` using a checksum with an algorithm different than sha1, `Verify` will complain that the reference is invalid even when it is well formed and [it is a valid checksum algorithm](https://spdx.github.io/spdx-spec/4-file-information/#44-file-checksum).

Example using ext reference 
`DocumentRef-kubernetes-v1.22.0-beta.1 https://k8s.io/sbom/source/v1.22.0-beta.1 SHA256: 9b01bf4e25f170cba6b8ad1db4297d2d4b7ddebbebf87cc440a1a6f537f48637`

```
java -jar tmp/tools-java-1.0.1-jar-with-dependencies.jar Verify tmp/kubernetes-release.spdx 
WARNING: sun.reflect.Reflection.getCallerClass is not supported. This will impact performance.
ERROR StatusLogger No log4j2 configuration file found. Using default configuration: logging only errors to the console. Set system property 'log4j2.debug' to show Log4j2 internal initialization logging.
Analysis exception processing SPDX file: Invalid external document reference: DocumentRef-kubernetes-v1.22.0-beta.1 https://k8s.io/sbom/source/v1.22.0-beta.1 SHA256: 9b01bf4e25f170cba6b8ad1db4297d2d4b7ddebbebf87cc440a1a6f537f48637 at line number 6
```

If I change the external reference to SHA1, using the following tag it works:
`ExternalDocumentRef:DocumentRef-kubernetes-v1.22.0-beta.1 https://k8s.io/sbom/source/v1.22.0-beta.1 SHA1: 89b0b411c5381564689f5ce96a84b2987d25acfa`

```
java -jar tmp/tools-java-1.0.1-jar-with-dependencies.jar Verify tmp/kubernetes-release.spdx 
WARNING: sun.reflect.Reflection.getCallerClass is not supported. This will impact performance.
ERROR StatusLogger No log4j2 configuration file found. Using default configuration: logging only errors to the console. Set system property 'log4j2.debug' to show Log4j2 internal initialization logging.
This SPDX Document is valid.
```

I also tested with sha512 and it also did not work.

**Comment by goneall** [comment]
@puerco my interpretation of the spec is that the algorithm must be SHA1, although it is a little unclear.  The Checksum for the ExternalDocumentRef refers to [section 4.4](https://spdx.github.io/spdx-spec/4-file-information/#44-file-checksum) which includes the text `4.4.4 Algorithm: SHA1is to be used on the file. Other algorithms that can be provided optionally include SHA224, SHA256, SHA384, SHA512, MD2, MD4, MD5, MD6`.

@swinslow How do you handle this in the Go libraries?  I recall having a similar discussion after the first NTIA plugfest, but I don't recall the details.

**Comment by puerco** [comment]
Ah yes thank you @goneall ! I see what you mean. I understood that bit as only applying to `Filechecksum:`. Funny enough I stumbled into this when following this example from @swinslow that uses SHA256: 

https://github.com/swinslow/spdx-examples/blob/5e898e3afa82592cdd6288a7a8f6464caf6f248f/example5/spdx/example5-bin.spdx#L1-L10

**Comment by swinslow** [comment]
@puerco Yes -- sorry about that! The examples in https://github.com/swinslow/spdx-examples were done before @goneall and I had discussed this ambiguity in the spec. Based on the discussion at https://github.com/swinslow/cmake-spdx/issues/8 it looks like I need to switch the examples to SHA1 instead of SHA256. I've now filed an issue at https://github.com/swinslow/spdx-examples/issues/31 to track this  :)

@goneall in the [SPDX Golang tools](https://github.com/spdx/tools-golang/), I was actually just looking at this again over the weekend. The Golang tools permit a document creator to use any of the checksum types for External Document Refs. That said, there isn't anything in the Golang tools that automatically creates an External Document Reference; so at this point I decided to leave it as possible to represent them with any of the checksum types. You can see my thinking at https://github.com/spdx/tools-golang/issues/67 -- and feel free to reopen or weigh in if you think I'm off-base there...

**Comment by goneall** [comment]
Thanks @swinslow   

I think a good course for the Java libraries is to leave the check in for Verify, but change it from an error to a warning for the other features of the tool.  

For reference, the verify check is at this line: https://github.com/spdx/Spdx-Java-Library/blob/293698c84e5d9390d3c3c91c4a6958245e2fadaf/src/main/java/org/spdx/library/model/ExternalDocumentRef.java#L293

I'll need to research where the error is being generated and change that to a warning.

**Comment by goneall** [comment]
The issue is in the spdx-java-tag-value-store - transferring the issue.
